cpp/src/arrow/table_test.cc

```diff
@@ -152,38 +153,234 @@ TEST_F(TestTable, AllColumnsAndFields) {
   ASSERT_EQ(0, fields.size());
 }
 
-TEST_F(TestTable, Equals) {
-  const int length = 100;
-  MakeExample1(length);
+class TestTableEquality : public TestTable {};
 
-  table_ = Table::Make(schema_, columns_);
+TEST_F(TestTableEquality, Equals) {
```

Comment:
It seems that we don't need to inherit `TestTable` because `TestTableEquality` doesn't use `MakeExample1()`.

How about using `TEST(TestTableEquality, Equals)`?